packages/core/src/internal/fetchSource.ts

```diff
@@ -57,21 +24,44 @@ export const makeFetchSource = (
           fetchOptions.signal = abortController.signal;
         }
 
-        return executeFetch(operation, url, fetchOptions);
+        // Does the fetch call — handling any network level errors, like not 200's
+        return fetcher(url, fetchOptions).then((response: Response) => {
+          catch_response = response;
+
+          if (
+            response.status < 200 ||
+            response.status >= (fetchOptions.redirect === 'manual' ? 400 : 300)
+          ) {
+            throw new Error(response.statusText);
+          }
+
+          return response;
+        });
       })
-      .then((result: OperationResult | undefined) => {
-        if (!ended) {
-          ended = true;
-          if (result) next(result);
-          complete();
-        }
-      });
+      // now we handle the maybe response
+      .then(async (response: Response | undefined) => {
+        if (!response || ended) return;
+
+        const networkResult = await response.json();
+
+        if (!('data' in networkResult) && !('errors' in networkResult))
+          throw new Error('No Content');
+
+        if (ended) return;
+
+        const result = makeResult(operation, networkResult, response);
+        if (result) next(result);
+      })
+      .catch((error: Error) => {
+        if (error.name === 'AbortError') return;
+
+        next(makeErrorResult(operation, error, catch_response));
+      })
+      .finally(() => complete());
```

Comment:
`.finally` is correctly polyfilled yeah?

Reply:
It's in fact not, and dynamic transpilation of methods is always a hairy situation, so I'd opt for not doing this here, personally. This will likely also become clearer as we add the actual multipart logic into this file, but generally, I think it isn't a bad thing to make sure that whatever uses `makeResult` also calls `next(result)` _and_ `complete()` as needed, since in multipart responses we can clearly tell when we're done.

For errors we can also always call it eagerly as we know we're done anyway